packages/mattermost-redux/src/selectors/entities/channels.ts

```diff
@@ -276,6 +276,29 @@ export const getCurrentChannel: (state: GlobalState) => Channel = createSelector
     },
 );
 
+export const getCurrentChannelNameForSearchShortcut: (state: GlobalState) => string | undefined = createSelector(
+    getAllChannels,
+    getCurrentChannelId,
+    (state: GlobalState): UsersState => state.entities.users,
+    getTeammateNameDisplaySetting,
+    (allChannels: IDMappedObjects<Channel>, currentChannelId: string, users: UsersState, teammateNameDisplay: string): string | undefined => {
+        const channel = allChannels[currentChannelId];
+
+        // Only get the extra info from users if we need it
+        if (channel?.type === Constants.DM_CHANNEL) {
+            const dmChannelWithInfo = completeDirectChannelInfo(users, teammateNameDisplay, channel);
+            return `@${dmChannelWithInfo.display_name}`;
+        }
+
+        // Replace spaces in GM channel names
+        if (channel?.type === Constants.GM_CHANNEL) {
+            return `@${channel.display_name.replace(/\s/g, '')}`;
```

Comment:
Ditto here. Also, what happens if the GM channel name exceeds the max length? That said, I'm not actually sure if the `in:` search handles that to begin with

Reply:
This one does work, because the logic to replace those names is elsewhere. We use this exact same logic in the search provider to get GMs. It's very confusing and inconsistent tbh :P